File: Biblioteca-de-Trabajo/3. CODIGO/Prototipo v1.0/src/negocio/GestorEventos.java
package negocio;

import java.util.ArrayList;
import java.util.List;

public class GestorEventos {
    private List<IObservadorAuditoria> observadores = new ArrayList<>();

    public void suscribir(IObservadorAuditoria obs) {
        observadores.add(obs);
    }

    public void notificar(String usuarioId, String rol, String accion, String detalles) {
        for (IObservadorAuditoria obs : observadores) {
            obs.registrar(usuarioId, rol, accion, detalles);
        }
    }
}

File: Biblioteca-de-Trabajo/3. CODIGO/Prototipo v2.0/src/negocio/GestorEventos.js
export class GestorEventos {
  constructor() {
    this.observadores = [];
  }

  suscribir(observer) {
    this.observadores.push(observer);
  }

  notificar(evento) {
    this.observadores.forEach((obs) => obs.actualizar(evento));
  }
}
